### PR TITLE
Thorough refactoring

### PR DESCRIPTION
**Pull request by padraigfl**
A collection of changes to figure out just what these pieces can be used for

- [x] ~consistent naming for functions across ssa and srt where possible~ (achieved via default exports)
- [x] test with actual files to try and find edge cases
- [x] ~test with actual files using highly popular non latin characters to see if viable~ (require UTF8 format strings)
- [ ] ~object attributes match with other popular subtitling modules~ (added to separate PR)
- [x] minimising codependency between operations 
(e.g. styling should not be necessary to use convert)
- [ ] ~just better names, clear names, damnit~

### CODE REVIEW

**Comment by padraigfl** [comment]
Use UTF-8 and depend on other modules and/or users to provide UTF-8 formatted files

File: src/pairing/index.js [hunk 1]
```diff
@@ -70,14 +53,14 @@ function makePairs(s1, s2, offset) {
...
-
   while (s1count < s1.length) {
     while (s2count < s2.length) {
       temp = isValidPair(s1[s1count], s2[s2count], offset);
-
       if (temp === c.FULL_MATCH || temp === c.START_LATE_END_EARLY || temp === c.START_EARLY_END_LATE) {
         primary[s1count].secondaryText = s2[s2count].text;
         break;
+      } else if (temp === c.START_AFTER_END) {
+        break;
```
Comment (padraigfl) [inline src/pairing/index.js:67]: possible option to use this to include total outliers?

File: src/io/index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@
 var fs = require('fs');
 
 function readFile(filename) {
-  return fs.readFileSync(filename, 'utf8', function(err, data) {
+  return fs.readFileSync(filename, 'ascii', function(err, data) {
```
Comment (padraigfl) [inline src/io/index.js]: revert back to utf8, beyond scope of project

File: src/pairing/constants.js [hunk 3]
```diff
@@ -13,4 +13,6 @@ module.exports = {
 
   START_EARLY_END_MATCH: 'START_EARLY_END_MATCH', // treat as within // future: double up?
   START_LATE_END_MATCH: 'START_LATE_END_MATCH', // treat as within for now
+
+  START_AFTER_END: 'START_AFTER_END',
```
Comment (padraigfl) [inline src/pairing/constants.js:17]: add END_BEFORE_START to catch all cases

File: src/ssa/parse.js [hunk 4]
```diff
@@ -69,7 +70,7 @@ function parseSsa(data) {
 module.exports = {
```
Comment (padraigfl) [inline src/ssa/parse.js:87]: SSA can do same exports as SRT, but lets leave that until the module is broken out on its own

File: src/ssa/styles/everything.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "font": "Times New Roman",
```
Comment (padraigfl) [inline src/ssa/styles/everything.json:2]: better defaults, allow opaque background

File: src/srt/constants/regEx.js [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
-var SIMPLE = /<\/?\w(\s*)>/;
-var FONT = /(<\/?font\s+color=('|")#?\w+('|")?>)|(<\/?font>)$/;
-var FULL = /<\/?\w+(\s+color=('|")(\w+)('|"))?>/;
+var SIMPLE = /<\/?\w\s*>/;
+var FONT = /^<\/?font\s+color=['"]#?\w+['"]>|<\/?font>$/;
+var FULL = /<\/?\w+\s*?c?o?l?o?r?=?['"\w]*?>/;
 var FIND_TAGS = /<.*>.*<\/.*>/;
-var VALIDATOR = /\d+\n((\d\d:\d\d:\d\d,\d\d\d\s*-->\s*\d\d:\d\d:\d\d,\d\d\d\n)([^\n]+\n)?([^\n]+\n))+/;
-
+var VALIDATOR = /\d+(\r\n|\n)\d\d:\d\d:\d\d,\d\d\d\s*-->\s*\d\d:\d\d:\d\d,\d\d\d(\r\n|\n)([^\n]+\n)+(\r\n|\n)*/;
```
Comment (padraigfl) [inline src/srt/constants/regEx.js:7]: move into relevant files

**Review by padraigfl — COMMENTED** [review]
and double check the name uniformity
